buddy_alloc.h

```diff
@@ -1591,7 +1599,7 @@ static float buddy_tree_fragmentation(struct buddy_tree *t) {
         size_t pos_status = buddy_tree_status(t, state.current_pos);
         if (pos_status == 0) {
             // Empty node, process
-            size_t virtual_size = 1u << ((tree_order - state.current_pos.depth) % ((sizeof(size_t) * CHAR_BIT)-1));
+            size_t virtual_size = 1ul << ((tree_order - state.current_pos.depth) % ((sizeof(size_t) * CHAR_BIT)-1));
```

Comment:
Interesting, what's causing this difference between C and C++ here ? I have rather limited knowledge of cpp - might be obvious but I don't know.